src/api/spec/controllers/webui/cloud/ec2/upload_job/logs_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe Webui::Cloud::UploadJob::LogsController, type: :controller, vcr: true do
+  let!(:user) { create(:confirmed_user, login: 'tom') }
+  let!(:ec2_configuration) { create(:ec2_configuration, user: user) }
+  let(:project) { create(:project, name: 'Apache') }
+  let!(:package) { create(:package, name: 'apache2', project: project) }
+
+  before do
+    login(user)
+  end
+
+  describe '#show' do
+    context 'with cloud_upload feature enabled' do
+      context 'without an EC2 configuration' do
+        before do
+          user.ec2_configuration = nil
+          user.save
```

Comment:
This shouldn't be needed. And if it's needed it probably would be better to define a :user_with_ec2_conf factory.

Reply:
@bgeuken it is needed, but I think it should be done differently. You don't need a factory for that, as it is just a normal user.

I would just have two users: 

```ruby
 let!(:user) { create(:confirmed_user, login: 'tim') }
 let(:ec2_configuration) { create(:ec2_configuration, user: user) }
 let!(:user_with_ec2_configuration) { create(:confirmed_user, login: 'tom', ec2_configuration: ec2_configuration) }
```

well, you may need to remove the `!` and create only one of the two user in every context.